### PR TITLE
attempt to fix asm parsing

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
semarie confirmed with me the issue on OpenBSD, due to overlapping buffer.

**Comment by radare** [comment]
1 XX

File: libr/parse/parse.c [hunk 1]
```diff
@@ -325,9 +327,15 @@ static int filter(RParse *p, RFlag *f, char *data, char *str, int len, bool big_
...
 								memmove (ptr_left, ptr_esc, copied_len);
-								sprintf (ptr_left + copied_len, "%s%s",
-										ansi_found && ptr_right - ptr_end + 1 >= 4 ? "\x1b[0m" : "",
-										ptr_right + 1);
+								dptr_left = strcpy (ptr_left + copied_len,
+										ansi_found && ptr_right - ptr_end + 1 >= 4 ? "\x1b[0m" : "");
+								int dlen = strlen (dptr_left);
+								dptr_left += dlen;
+								dptr_end = ptr_right + 1;
+								while (*dptr_end && ++dptr_end != '\0');
```
Comment (radare) [inline libr/parse/parse.c]: this is an overengineered strlen

File: libr/parse/parse.c [hunk 2]
```diff
@@ -325,9 +327,15 @@ static int filter(RParse *p, RFlag *f, char *data, char *str, int len, bool big_
 									break;
 								}
 								memmove (ptr_left, ptr_esc, copied_len);
-								sprintf (ptr_left + copied_len, "%s%s",
-										ansi_found && ptr_right - ptr_end + 1 >= 4 ? "\x1b[0m" : "",
-										ptr_right + 1);
+								dptr_left = strcpy (ptr_left + copied_len,
+										ansi_found && ptr_right - ptr_end + 1 >= 4 ? "\x1b[0m" : "");
```
Comment (radare) [inline libr/parse/parse.c]: so much scary pointer arithmetics here

File: libr/parse/parse.c [hunk 3]
```diff
@@ -325,9 +327,15 @@ static int filter(RParse *p, RFlag *f, char *data, char *str, int len, bool big_
...
+								int dlen = strlen (dptr_left);
+								dptr_left += dlen;
+								dptr_end = ptr_right + 1;
+								while (*dptr_end && ++dptr_end != '\0');
+								int llen = (dptr_end - (ptr_right + 1));
```
Comment (radare) [inline libr/parse/parse.c]: unnecessary parenthesis

File: libr/parse/parse.c [hunk 4]
```diff
@@ -325,9 +327,15 @@ static int filter(RParse *p, RFlag *f, char *data, char *str, int len, bool big_
 									break;
 								}
 								memmove (ptr_left, ptr_esc, copied_len);
-								sprintf (ptr_left + copied_len, "%s%s",
-										ansi_found && ptr_right - ptr_end + 1 >= 4 ? "\x1b[0m" : "",
-										ptr_right + 1);
+								dptr_left = strcpy (ptr_left + copied_len,
```
Comment (radare) [inline libr/parse/parse.c]: define and assign the variables here, no need to define them on top of the while

File: libr/parse/parse.c (hunk 2 above)
Comment (radare) [reply libr/parse/parse.c]: do not hardcode ansi escapecodes. this is not portable. use Color_RESET

Comment (radare) [reply libr/parse/parse.c]: add parenthesis for this ternary conditional. or the >=, - and && operation may behave wrong in some compilers

**Comment by devnexen** [comment]
UT pass with last commit.

**Pull request merged**